ApiData.ts

```diff
@@ -0,0 +1,209 @@
+/** class to manage data and setting on the grid */
+class ApiData {
+	// Properties to manage data and settings
+	pageSize: number;
+	currentPage: number = 1;
+	data: GridData[] = [];
+	totalItems: number = 0;
+	columnNames: ColumnName[] = [];
+	maxGridHeight: number = 0;
+	firstVal: number = 0;
+	lastVal: number = -1;
+
+	constructor(pageSize: number) {
+		this.pageSize = pageSize;
+	}
+
+	/** Initialize method to set up the grid */
+	async initialize(): Promise<void> {
+		Promise.resolve()
+			.then(() => this.adjustGridHeight())
+			.then(() => this.recordCount())
+			.then(() => this.fetchColumns())
```

Comment:
Please don't create a `Promise.resolve` just to chain with another promise :P. And `adjustGridHeight` does not return a promise, so no need to treat it as such. And please return the promise to allow other function to await on this function in case they need to.

```suggestion
		this.adjustGridHeight();
		return this.recordCount()
			.then(() => this.fetchColumns())
```